keras/src/callbacks/orbax_checkpoint_test.py

```diff
@@ -577,3 +587,801 @@ def compare_nested_dicts(orig_dict, loaded_dict):
+                    axes=(None, "data")
+                )
+                layout_map["output_layer/bias"] = TensorLayout(axes=(None,))
+            else:
+                # Single device distribution
+                device_mesh = DeviceMesh((1,), axis_names=["data"])
+                layout_map = LayoutMap(device_mesh)
+                layout_map["dense_layer/kernel"] = TensorLayout(
+                    axes=(None, None)
+                )
+                layout_map["dense_layer/bias"] = TensorLayout(axes=(None,))
+                layout_map["output_layer/kernel"] = TensorLayout(
+                    axes=(None, None)
+                )
+                layout_map["output_layer/bias"] = TensorLayout(axes=(None,))
+
+            distribution = ModelParallel(
+                device_mesh=device_mesh, layout_map=layout_map
+            )
+
+            # Save original distribution state
+            original_distribution = None
+            try:
+                from keras.src.distribution import (
+                    distribution as get_distribution,
+                )
+
+                original_distribution = get_distribution()
+            except (ImportError, AttributeError):
+                pass
+
+            try:
+                # Set distribution
+                set_distribution(distribution)
+
+                # Create model with distribution
+                model = self._create_test_model()
+                x, y = self._create_dummy_data()
+
+                checkpoint_dir = os.path.join(
+                    self.get_temp_dir(), "test_resharding"
+                )
+                callback = OrbaxCheckpoint(
+                    directory=checkpoint_dir, save_freq="epoch"
+                )
+
+                # Train and save with original distribution
+                model.fit(x, y, epochs=2, callbacks=[callback], verbose=0)
+                callback.wait_until_finished()
+
+                # Create new model and load with same distribution
+                new_model = self._create_test_model()
+                # Initialize optimizer state by running a dummy training step
+                batch_size = min(2, len(x))  # Compatible with distribution
+                new_model.fit(
+                    x[:batch_size], y[:batch_size], epochs=0, verbose=0
+                )
+
+                # Get initial weights before loading
+                initial_weights = new_model.get_weights()
+
+                new_model.load(checkpoint_dir)
+                loaded_weights = new_model.get_weights()
+
+                # Get original weights for comparison
+                original_weights = model.get_weights()
+
+                # Check that loading actually changed some weights
+                loading_changed_weights = any(
+                    not np.allclose(init, loaded)
+                    for init, loaded in zip(initial_weights, loaded_weights)
+                )
+                self.assertTrue(
```

Comment:
Don't use `assertTrue`, it won't tell you which one if off, by how much, etc.

Reply:
Done